stuff/dataset_generator/runner.py

```diff
@@ -0,0 +1,188 @@
+"""
+Local request runner compatible with bootstrap signature.
+"""
+
+from __future__ import annotations
+
+import logging
+import time
+from typing import Any, Dict, Mapping, Optional, cast
+import yaml
+
+SETTINGS_PATH = "./stuff/dataset_generator/settings.yml"
+
+
+def _normalize_now(params: Dict[str, Any], now_ts: float) -> None:
+    """
+    replace $NOW$[+/-offset] tokens in-place with epoch seconds.
+    """
+    for key, val in list(params.items()):
+        if isinstance(val, str) and val.startswith("$NOW$"):
+            expr = val.replace("$NOW$", "", 1).strip()
+            offset = 0
+            if expr:
+                try:
+                    offset = int(expr)
+                except ValueError:
+                    offset = 0
+            params[key] = int(now_ts + offset)
+
+
+def _normalize_key(endpoint: str) -> str:
+    """
+    Normalize an endpoint for dictionary
+    lookups (lowercase + trailing slash).
+    """
+    normalized = endpoint.strip().lower()
+    return normalized if normalized.endswith("/") else normalized + "/"
+
+
+def _ensure_slash(endpoint: str) -> str:
+    """
+    Ensure trailing slash for real endpoints
+    while preserving case.
+    """
+    normalized = endpoint.strip()
+    return normalized if normalized.endswith("/") else normalized + "/"
+
+
+def _deep_get(
+    mapping: Mapping[str, object],
+    path: str
+) -> object | None:
+    """Return a nested value from a mapping using a dot-separated path."""
+    current: object = mapping
+    for key in path.split("."):
+        if not isinstance(current, Mapping) or key not in current:
+            return None
+        current = current[key]  # type: ignore[index]
+    return current
+
+
+with open(SETTINGS_PATH, "r", encoding="utf-8") as f:
+    _SETTINGS: dict[str, object] = yaml.safe_load(f) or {}
+
+_ENDPOINTS: Mapping[str, str] = _SETTINGS.get(
+    "endpoints", {}
+)  # type: ignore[assignment]
+_RULES: list[dict[str, object]] = _SETTINGS.get(
+    "resource_checks", []
+)  # type: ignore[assignment]
+
+_RESOURCE_CHECKS: dict[str, dict[str, object]] = {}
+for rule in _RULES:
+    if not isinstance(rule, dict):
+        continue
+
+    create_key = rule.get("create")
+    if not isinstance(create_key, str):
+        continue
+
+    create_endpoint = _ENDPOINTS.get(create_key)
+    if not create_endpoint:
+        continue
+
+    _RESOURCE_CHECKS[_normalize_key(create_endpoint)] = {
+        "check_api": _ensure_slash(str(rule.get("check_api", ""))),
+        "params_map": rule.get("params_map", {}),
+        "expect": rule.get("expect", []),
+    }
```

Comment:
Acá ya podrías utilizar la `dataclass` :)
```suggestion
    _RESOURCE_CHECKS[_normalize_key(create_endpoint)] = ResourceCheck(
        check_api=_ensure_slash(str(rule.get("check_api", ""))),
        params_map=rule.get("params_map", {}),
        expect=rule.get("expect", []),
    )
```